Use range-for instead of llvm::for_each over std::unordered_set ranges: with GCC 11 in C++20
mode the call fails with "no matching function for call to adl_begin". Not reported upstream.

diff --git intel-graphics-compiler/IGC/VectorCompiler/lib/GenXCodeGen/GenXStructSplitter.cpp intel-graphics-compiler/IGC/VectorCompiler/lib/GenXCodeGen/GenXStructSplitter.cpp
index 334462e73..7446d1c65 100644
--- intel-graphics-compiler/IGC/VectorCompiler/lib/GenXCodeGen/GenXStructSplitter.cpp
+++ intel-graphics-compiler/IGC/VectorCompiler/lib/GenXCodeGen/GenXStructSplitter.cpp
@@ -842,15 +842,16 @@ void DependencyGraph::processNode(Node &SNode) {
       // In case of simplification: type will be substituted directly, without
       // structure.
       // F { float } C { F, float } C_BS -> { float , float }
-      llvm::for_each(SNode.parents(), [&](Node *ParentNode) {
+      for (Node *ParentNode : SNode.parents()) {
         remakeParent(*ParentNode, SNode, GeneratedTypesInOrder);
-      });
+      }
     }
   }
 
   // Removes dependencies.
-  llvm::for_each(SNode.parents(),
-                 [&SNode](Node *ParentNode) { ParentNode->eraseChild(SNode); });
+  for (Node *ParentNode : SNode.parents()) {
+    ParentNode->eraseChild(SNode);
+  }
 }
 
 //
